src/cdk/overlay/overlay-directives.ts

```diff
@@ -340,6 +345,8 @@ export class ConnectedOverlayDirective implements OnDestroy, OnChanges {
     this._position.withDirection(this.dir);
     this._overlayRef.getState().direction = this.dir;
     this._initEscapeListener();
+    /** Update the overlay size, in case the directive's inputs have changed */
+    this._configureSize(this._overlayRef.getState());
```

Comment:
The `getState()` method isn't really meant to be used to set state. @jelbourn and I discussed and we think there should be a dedicated method in `OverlayRef` to set the size props in the state. There is already an `updateSize` method, but it's really an internal implementation. So let's:

- rename the existing `updateSize` to `_setSizeStyles`
- add a new `updateSize` method that simply updates the size props in the state

Then you can use the new `updateSize` to set the props here, instead of `getState()`.